app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.java

```diff
@@ -48,4 +49,36 @@ protected void updateDefaultItem(CompoundButton defaultButton) {
         }
         return engineIdSet;
     }
+
+    // Whenever an engine is checked or unchecked, we adapt the MenuItem state (enable or disable) given in param
+    // Though some conditions are added just for optimisation
+    public void bindEnginesCheckboxesTo(final MenuItem menuDeleteItem, final int alpha_default, final int alpha_greyed) {
+        for (int i = 0; i < searchEngineGroup.getChildCount(); i++) {
+            final CompoundButton engineButton = (CompoundButton) searchEngineGroup.getChildAt(i);
+            engineButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                    if (buttonView.isChecked() || atLeastOneEngineChecked()) {
```

Comment:
Nit: this first buttonView.isChecked can just be the isChecked boolean in the arguments.

Reply:
OMG didn't even noticed that param :) . Thank you